src/00b3_survey_mics/script.R

```diff
@@ -72,94 +88,144 @@ mics_paths <- load_sharepoint_data(
+     mutate(analysis = "circ", dataset = "mn") %>% # valid??
+     # don't have female date of births or interviews
+     filter(!var_raw %in% c("wdoi", "wdob")))
+  ) %>%
   # add in default row for mics_areas (hh7) for dataset == mn
   # also add rows for date of birth and date of interview (two different forms)
   bind_rows(
     tibble::tribble(
       ~survey_id,      ~dataset, ~variable,        ~var_raw,  ~var_label_raw,                ~analysis,
       "_default_mics", "mn",     "mics_area_name", "hh7",     "MICS area level",             "circ",
-      "ZWE2014MICS", "mn",     "dob",            "mwdob",   "Date of birth of man (CMC)",  "circ", 
-      "_default_mics", "mn",     "dob",            "mdob",    "Date of birth of man (CMC)",  "circ",
-      "ZWE2014MICS", "mn",     "doi",            "mwdoi",   "Date of interview men (CMC)", "circ",
-      "_default_mics", "mn",     "doi",            "mdoi",    "Date of interview men (CMC)", "circ"
+      "SWZ2010MICS",   "mn",     "dob",            "mdob",    "Date of birth of man (CMC)",  "circ",
+      "_default_mics", "mn",     "dob",            "mwdob",   "Date of birth of man (CMC)",  "circ",
+      "SWZ2010MICS",   "mn",     "doi",            "mdoi",    "Date of interview men (CMC)", "circ",
+      "_default_mics", "mn",     "doi",            "mwdoi",   "Date of interview men (CMC)", "circ",
     )
   )
 
+#' CAF2006MICS adjustments:
+#' * Dataset is "mm" (instead of "mn")
+#' * DOB variable is `wdob`
+#' * circ_status is `hha19`
+
+variable_recode <- variable_recode %>%
+  mutate(dataset = if_else(survey_id == "CAF2006MICS" & analysis == "circ", "mm", dataset)) %>%
+    bind_rows(
+    tibble::tribble(
+      ~survey_id,      ~dataset, ~variable,        ~var_raw,  ~var_label_raw,                ~analysis,
+      "CAF2006MICS",   "mm",     "circ_status",    "hha19",   "Etes vous circoncis",         "circ",
+      "CAF2006MICS",   "mm",     "dob",            "wdob",    "Date of birth (CMC)",         "circ",
+      "CAF2006MICS",   "mm",     "mnweight",       "meweight", "[Sampling weight]",          "circ",
+      "CAF2006MICS",   "mm",     "cluster_id",     "hh1",     "Cluster ID - man dataset",      NA,
+      "CAF2006MICS",   "mm",     "household",      "hh2",     "Household ID - man dataset",    NA,
+      "CAF2006MICS",   "mm",     "line",           "ln",      "Line number - man dataset",     NA
+      )
+  )
+
+
+
 # extract survey variables, recode ID variables in in MICS surveys
 circ_extracted <- Map(
   extract_survey_vars,
   df = circ_raw,
-  survey_id = names(circ_raw),
-  list(variable_recode),
-  file_type[names(circ_raw)],
-  analysis = "circ",
-  # want to keep, but not recode 
-  add_vars = list(c("mnweight")))
+  survey_id_c = names(circ_raw),
+  variable_recode = list(variable_recode),
+  dataset_type = mics_male_dataset[names(circ_raw)],
+  analysis_c = "circ",
+  # want to keep, but not recode
+  add_vars = list(c("mnweight", "meweight", "mwm6y", "mwm6m", "mwm6d"))
```

Comment:
* Handling of the weight variable here seems a bit clunky for cases where it takes different names in different data sets. Seems to me like it would be better to handle the weight variable as a recode variable? Is there a reason I'm not thinking of that it is not done that way? 

* The other `mwm6*` variables I added here were custom export from the Nigeria survey for the date of interview to convert to a CMC interview date, which was not included in the male data set. This seems like a reasonable workflow to me to handle cases where a custom variable is needed:
  * extract additional variables
  * manually do calculated variables required
  * then drop the unneeded custom variables